fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java

```diff
@@ -169,35 +175,39 @@ public static Pair<List<Expression>, List<Expression>> extractExpressionForHashT
      * Get all used slots from onClause of join.
      * Return pair of left used slots and right used slots.
      */
-    public static Pair<List<SlotReference>, List<SlotReference>> getOnClauseUsedSlots(
-            AbstractPhysicalJoin<Plan, Plan> join) {
-        Pair<List<SlotReference>, List<SlotReference>> childSlots =
+    public static Pair<List<ExprId>, List<ExprId>> getOnClauseUsedSlots(
+                AbstractPhysicalJoin<? extends Plan, ? extends Plan> join) {
+        Pair<List<ExprId>, List<ExprId>> childSlotsExprId =
                 Pair.of(Lists.newArrayList(), Lists.newArrayList());
 
         List<SlotReference> leftSlots = Utils.getOutputSlotReference(join.left());
         List<SlotReference> rightSlots = Utils.getOutputSlotReference(join.right());
-        List<EqualTo> equalToList = join.getHashJoinConjuncts().stream()
+        List<EqualTo> equalToList =  join.getHashJoinConjuncts().stream()
```

Comment:
strange things.
Why `checkstyle` doesn't check this space.

Reply:
add a rule to check it in another PR #12354